tensorflow/python/compiler/tensorrt/trt_convert.py

```diff
@@ -1196,6 +1197,16 @@ def convert(self, calibration_input_fn=None):
     """
     assert not self._converted
 
+    if self._device is None:
+      # Creating an empty tensor to fetch queried device
+      self._device = array_ops.zeros([]).device
+      if not self._device:  # if device unspecified, defaulting to GPU 0
+        self._device = "gpu:0"
+      if "gpu" not in self._device.lower():
+        raise ValueError(f"Specified device is not a GPU: {self._device}")
+      logging.info(f"Placing imported graph from "
+                   f"`{self._input_saved_model_dir}` on device: {self._device}")
```

Comment:
Should this be in the same indentation level as line 1200?

Reply:
Good point -- doing so will log it for all cases.

Reply:
@bixia1  -- done. Please review!